graylog2-plugin-interfaces/src/main/java/org/graylog2/plugin/KafkaJournalConfiguration.java

```diff
@@ -32,6 +33,25 @@
 
 public class KafkaJournalConfiguration {
 
+    public KafkaJournalConfiguration() { }
+
+    @JsonCreator
+    public KafkaJournalConfiguration(@JsonProperty("directory") File messageJournalDir,
+                                     @JsonProperty("segment_size") long messageJournalSegmentSize,
+                                     @JsonProperty("segment_age") Duration messageJournalSegmentAge,
+                                     @JsonProperty("max_size") long messageJournalMaxSize,
+                                     @JsonProperty("max_age") Duration messageJournalMaxAge,
+                                     @JsonProperty("flush_interval") long messageJournalFlushInterval,
+                                     @JsonProperty("flush_age") Duration messageJournalFlushAge) {
+        this.messageJournalDir = messageJournalDir;
```

Comment:
This should probably be guarded by [`Objects#requireNonNull()`](https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#requireNonNull-T-) as it's a required value.